plugins/destination/mongodb/client/deletestale.go

```diff
@@ -0,0 +1,19 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+func (c *Client) DeleteStale(ctx context.Context, tables schema.Tables, source string, syncTime time.Time) error {
+	for _, table := range tables {
```

Comment:
This probably could/should be done concurrently (easy enough with an `errgroup`)